opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -0,0 +1,418 @@
+            {
+                return cell_idx_;
+            }
+
+            inline const int& aquiferID() const
+            {
+                return aquiferID_;
+            }
+
+
+        private:
+            const Simulator& ebos_simulator_;
+
+
+            // Aquifer ID, and other IDs
+            int aquiferID_, inftableID_, pvttableID_;
+
+            // Grid variables
+
+            std::vector<size_t> cell_idx_;
+            std::vector<Scalar> faceArea_connected_;
+
+            // Quantities at each grid id
+            std::vector<Scalar> cell_depth_;
+            std::vector<Eval> pressure_previous_;
+            std::vector<Eval> pressure_current_;
+            std::vector<Eval> Qai_;
+            std::vector<Eval> rhow_;
+            std::vector<Scalar> alphai_;
+
+            // Variables constants
+            Scalar mu_w_ , //water viscosity
+                   phi_aq_ , //aquifer porosity
+                   d0_,     // aquifer datum depth
+                   C_t_ , //total compressibility
+                   r_o_ , //aquifer inner radius
+                   k_a_ , //aquifer permeability
+                   c1_, // 0.008527 (METRIC, PVT-M); 0.006328 (FIELD); 3.6 (LAB)
+                   h_ , //aquifer thickness
+                   theta_ , //angle subtended by the aquifer boundary
+                   c2_ ; //6.283 (METRIC, PVT-M); 1.1191 (FIELD); 6.283 (LAB).
+
+            // Variables for influence table
+            std::vector<Scalar> aqutab_td_, aqutab_pi_;
+
+            // Cumulative flux
+            Scalar dt_, pa0_, gravity_;
+            bool p0_defaulted_;
+            Eval W_flux_;
+
+            
+            inline const double area_fraction(const size_t i)
+            {
+                return alphai_.at(i);
+            }
+
+            inline void get_influence_table_values(Scalar& pitd, Scalar& pitd_prime, const Scalar& td)
+            {
+                // We use the opm-common numeric linear interpolator
+                pitd = Opm::linearInterpolation(aqutab_td_, aqutab_pi_, td);
+                pitd_prime = Opm::linearInterpolationDerivative(aqutab_td_, aqutab_pi_, td);
+            }
+
+            inline void init_quantities(const Aquancon::AquanconOutput& connection)
+            {
+                // We reset the cumulative flux at the start of any simulation, so, W_flux = 0
+                W_flux_ = 0.;
+
+                // We next get our connections to the aquifer and initialize these quantities using the initialize_connections function
+                initialize_connections(connection);
+
+                calculate_aquifer_condition();
+
+                pressure_previous_.resize(cell_idx_.size(), 0.);
+                pressure_current_.resize(cell_idx_.size(), 0.);
+                Qai_.resize(cell_idx_.size(), 0.0);
+            }
+
+            inline void get_current_Pressure_cell(std::vector<Eval>& pressure_water, const int idx, const IntensiveQuantities& intQuants)
```

Comment:
please change the name of the function. You use the function get_current_Pressure_cell to set both the `pressure_current_` and `pressure_previous_`, which is really confusing. You can change the function name to be
something like `set_pressure_cell` or something. The function is more like doing `setting` other than `getting`.